$NetBSD: patch-an,v 1.2 2010/05/11 11:42:45 yhardy Exp $

SHA1_Update conflicts with openssl which may be dynamically loaded
at runtime via libcups or libgssapi so causing a crash due to using
the wrong binding.  So rename here to avoid conflict.

--- security/nss/lib/freebl/sha-fast-amd64-sun.s.orig	2009-06-29 18:15:14.000000000 +0200
+++ security/nss/lib/freebl/sha-fast-amd64-sun.s
@@ -1712,9 +1712,9 @@ shaCompress:
 .LFE7:
 	.size	shaCompress, .-shaCompress
 	.align 16
-.globl SHA1_Update
-	.type	SHA1_Update, @function
-SHA1_Update:
+.globl NSS_SHA1_Update
+	.type	NSS_SHA1_Update, @function
+NSS_SHA1_Update:
 .LFB5:
 	pushq	%rbp
 .LCFI5:
@@ -1800,7 +1800,7 @@ SHA1_Update:
 	call	shaCompress
 	jmp	.L245
 .LFE5:
-	.size	SHA1_Update, .-SHA1_Update
+	.size	NSS_SHA1_Update, .-NSS_SHA1_Update
 	.section	.rodata
 	.align 32
 	.type	bulk_pad.0, @object
@@ -1902,7 +1902,7 @@ SHA1_End:
 	subl	%r8d, %edx
 	andl	$63, %edx
 	incl	%edx
-	call	SHA1_Update@PLT
+	call	NSS_SHA1_Update@PLT
 	movq	%rbx, %rdi
 	movq	%r12, %rsi
 	shrq	$32, %rdi
@@ -2018,7 +2018,7 @@ SHA1_HashBuf:
 	movl	%r12d, %edx
 	movq	%r13, %rsi
 	movq	%rbx, %rdi
-	call	SHA1_Update@PLT
+	call	NSS_SHA1_Update@PLT
 	leaq	-292(%rbp), %rdx
 	movq	%r14, %rsi
 	movq	%rbx, %rdi
